epoch.go

```
@@ -1380,18 +1397,18 @@ func (e *Epoch) handleBlockMessage(message *BlockMessage, from NodeID) error {
 // processFinalizedBlocks processes a block that has a finalization certificate.
 // if the block has already been verified, it will index the finalization certificate,
 // otherwise it will verify the block first.
-func (e *Epoch) processFinalizedBlock(finalizedBlock *FinalizedBlock) error {
-	round, exists := e.rounds[finalizedBlock.FCert.Finalization.Round]
+func (e *Epoch) processFinalizedBlock(block Block, fCert FinalizationCertificate) error {
+	round, exists := e.rounds[fCert.Finalization.Round]
 	// dont create a block verification task if the block is already in the rounds map
 	if exists {
 		roundDigest := round.block.BlockHeader().Digest
-		seqDigest := finalizedBlock.FCert.Finalization.BlockHeader.Digest
+		seqDigest := fCert.Finalization.BlockHeader.Digest
```

I think I missed this before: 

```
if !bytes.Equal(roundDigest[:], seqDigest[:]) {
			e.Logger.Warn("Received finalized block that is different from the one we have in the rounds map",
				zap.Stringer("roundDigest", roundDigest), zap.Stringer("seqDigest", seqDigest))
			return nil
		}
```

But this shouldn't happen. we should return an error here instead, and mark the halted error.

---

done